pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -259,24 +263,15 @@ func TestPriorityQueue_AddUnschedulableIfNotPresent_Async(t *testing.T) {
 				},
 			},
 		}
-		cycle := q.SchedulingCycle()
-		go func() {
-			<-moveReqChan
-			q.AddUnschedulableIfNotPresent(unschedulablePod, cycle)
-			wg.Done()
-		}()
-	}
-	firstPod, _ := q.Pop()
-	if !reflect.DeepEqual(&expectedPods[0], firstPod) {
-		t.Errorf("Unexpected pod. Expected: %v, got: %v", &expectedPods[0], firstPod)
+
+		q.AddUnschedulableIfNotPresent(unschedulablePod, oldCycle)
 	}
-	// close moveReqChan here to make sure q.AddUnschedulableIfNotPresent is called after another pod is popped
-	close(moveReqChan)
-	wg.Wait()
-	// all other pods should be in active queue again
+
+	// Since there has been no move request or (effectively) any time passed since
```

Comment:
Actually, there has been a move request at "oldCycle" and that's why pods should be in the back off queue, otherwise they should have been in unschedulableQ.
In order to check the logic of this PR, we should also check that `isPodBackingOff` returns true for these pods.

Reply:
Thanks, I've corrected the comment.